tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/ServerUtil.java

```diff
@@ -92,6 +92,7 @@
 public class ServerUtil
 {
 
+    private final static Map<String,Object> sdkBundle = new HashMap<String,Object>();
```

Comment:
If we don't cache the bundle, we'll create a new bundle while the getPortalBundle was invoked every time. If we cache the bundle, then the jsp validate only spend 1s, if we don't cache, the validate will spend 3s.